backstop/src/emissions/manager.rs

```diff
@@ -1,26 +1,28 @@
 use cast::{i128, u64};
 use sep_41_token::TokenClient;
 use soroban_fixed_point_math::FixedPoint;
-use soroban_sdk::{panic_with_error, unwrap::UnwrapOptimized, vec, Address, Env, Vec};
+use soroban_sdk::{panic_with_error, unwrap::UnwrapOptimized, Address, Env};
 
 use crate::{
     backstop::{load_pool_backstop_data, require_pool_above_threshold},
-    constants::{BACKSTOP_EPOCH, SCALAR_7},
+    constants::{BACKSTOP_EPOCH, SCALAR_14, SCALAR_7},
     dependencies::EmitterClient,
     errors::BackstopError,
-    storage::{self, BackstopEmissionData},
+    storage::{self, BackstopEmissionData, RzEmissionData},
     PoolBalance,
 };
 
 use super::distributor::update_emission_data;
 
 /// Add a pool to the reward zone. If the reward zone is full, attempt to swap it with the pool to remove.
-pub fn add_to_reward_zone(e: &Env, to_add: Address, to_remove: Address) {
+pub fn add_to_reward_zone(e: &Env, to_add: Address, to_remove: Option<Address>) {
     let mut reward_zone = storage::get_reward_zone(e);
+    let rz_emission_index = storage::get_rz_emission_index(e);
     let max_rz_len = if e.ledger().timestamp() < BACKSTOP_EPOCH {
         10
     } else {
-        10 + (i128(e.ledger().timestamp() - BACKSTOP_EPOCH) >> 23) // bit-shift 23 is ~97 day interval
+        // Max reward zone length is 50
+        (10 + (i128(e.ledger().timestamp() - BACKSTOP_EPOCH) >> 23)).min(50) // bit-shift 23 is ~97 day interval
     };
```

Comment:
this should be fixed to 50, we can also get rid of the `BACKSTOP_EPOCH` variable